### ISSUE TITLE
[BUG]Unexplained crash

### ISSUE THREAD

**Issue by realericmc** [cagnulein/qdomyos-zwift#2305]
**Describe the bug**
Apologies, but I don't know what precipitated the crash.  I was a bit over an hour and 9 minutes into an elliptical session when it went down, and multiple repairing and restarts were needed to get it working again.

**To Reproduce**
Sole E35 elliptical (2023 version)
polar heart rate strap connected to E35 (and to kickr headwind)
qdomyos running on iPhone 15 pro max running iOS 17.4.1
zwift running on iPad Pro M2 running 17.4.1
qdomyos paired to zwift using virtual wifi for cadence, speed, and heart rate 

**Expected behavior**
no crash :-)

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: iPhone 15 pro max
 - OS: iOS
 - Version 17.4.1

**Append a debug log**

[debug-Wed_Apr_24_11_18_36_2024.log.zip](https://github.com/cagnulein/qdomyos-zwift/files/15098595/debug-Wed_Apr_24_11_18_36_2024.log.zip)
  
**Additional context**

**Comment by cagnulein** [comment]
do you also send the crash report when the iphone asked?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 24 apr 2024 alle 20:09 Eric McMurry ***@***.***>
ha scritto:

**Comment by realericmc** [comment]
it didn't ask to send the crash report.  Perhaps it is one of these:
[qdomyos analytics.zip](https://github.com/cagnulein/qdomyos-zwift/files/15098698/qdomyos.analytics.zip)

clock time, the crash should have been around 11:06 I think.  11:40 would be when I finished the run after restarting it

**Comment by cagnulein** [comment]
so the debug log that i need is the previous one, the one with the crash. i
will check on test flight if there is a crash

how often do you use qz? is it your first crash?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 24 apr 2024 alle 20:18 Eric McMurry ***@***.***>
ha scritto:

**Comment by realericmc** [comment]
yep, sorry about that, didn't dig into it enough.  here are the logs for up to and after the crash.
[Archive.zip](https://github.com/cagnulein/qdomyos-zwift/files/15098890/Archive.zip)

I use it a couple of times a week and it has crashed before, but not in the middle of a run.  that has happened before while trying to get all of the pairings going.

**Comment by cagnulein** [comment]
i will deeply check the log tomorrow but i have already found an issue on
your setup: since you have a bike and a elliptical on the same room you
have to force qz to connect to one or another. do it set the manual device
in the advanced settings. it should also fix your crashes

let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 24 apr 2024 alle 20:33 Eric McMurry ***@***.***>
ha scritto:

**Comment by realericmc** [comment]
thanks for the pointer.  I usually have that setting on but I do recall I had turned it off at some point before while going through the pairing dance.  I will keep it turned on.

**Comment by cagnulein** [comment]
got the issue! it was a malformed packet from the elliptical that caused this. I'm fixing right now!

**Comment by realericmc** [comment]
Cool.  That doesn’t surprise me, the sw on this elliptical seems like it could use a bit more testing.

Thanks!

**Comment by cagnulein** [comment]
Did you get the new version? Some users are struggling with testflight
these days

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 25 apr 2024 alle 15:08 Eric McMurry ***@***.***>
ha scritto:

> Cool. That doesn’t surprise me, the sw on this elliptical seems like it
> could use a bit more testing.
>
> Thanks!
>
>
>

**Comment by realericmc** [comment]
I have it.  It launches at least, but I haven’t tried it with the elliptical yet.  Thanks!

**Comment by cagnulein** [comment]
@realericmc any news?

**Comment by realericmc** [comment]
hi @cagnulein.  I think we're all good.  we had a lot of threads going and this was probably answered in one of the others.  sorry about that.  The last version seemed to work fine and didn't crash.  thanks!

**Comment by cagnulein** [comment]
Perfect, thanks
